src/components/header/Nav/Nav.tsx

```diff
@@ -0,0 +1,70 @@
+import { useState, useEffect } from 'react';
+import { getUser, getUserById } from '@/pages/api/api';
+import Profile from '../Profile/Profile';
+import styles from './Nav.module.css';
+import Link from 'next/link';
+import Image from 'next/image';
+
+interface Props {
+  className?: string | undefined;
+  setUserId?: (value: number) => void;
+  id?: number | undefined;
+}
+
+export interface User {
+  id: number;
+  name: string;
+  email: string;
+  profileImageSource?: string;
+  image_source?: string;
+}
+
+export default function Nav({ className = '', setUserId, id }: Props) {
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    async function applyGetUser() {
+      const nextUser = await getUser();
+      if (!nextUser) return;
+      setUser(nextUser);
+    }
+
+    async function apllyGetUserById(id: number) {
+      const nextUser = await getUserById(id);
+      if (!nextUser) return;
+      setUser(nextUser.data[0]);
+      if (setUserId) {
+        setUserId(nextUser.data[0].id);
+      }
+    }
+
+    if (id) apllyGetUserById(id);
+    else applyGetUser();

```

Comment:
위 코멘트 내용처럼 이렇게 중괄호 넣는 코딩스타일이 실수 방지에 좋습니다.
```suggestion
    if (id) {
       apllyGetUserById(id);
    } else { 
       applyGetUser();
    }
```